### ISSUE TITLE
Provide OSGI support

### ISSUE THREAD

**Issue by bjmi** [jspecify/jspecify#362]
JSpecify annotations could also be used in OSGI environments if `MANIFEST.MF` would have bundle entries like e.g.
```
Bundle-ManifestVersion: 2
Bundle-Name: JSpecify annotations
Bundle-SymbolicName: org.jspecify.annotations
Bundle-Version: 0.3.0
Export-Package: org.jspecify.annotations
[...]
```
Please provide a solution and JSpecify annotations could be used out of the box.

**Comment by kevinb9n** [comment]
I'm OSGIgnorant! If it would be easy enough to send us a PR please do, otherwise thanks for giving us the main idea.

**Comment by chrisrueger** [comment]
I will have a look if I can provide a PR

**Comment by chrisrueger** [comment]
> Can you summarize what the impact of this would be? What would that let people do that they can't do without it?
> 
As described in PR https://github.com/jspecify/jspecify/pull/428 

It helps and makes it easier for people to use jspecify in an Osgi environment. 

> If we don't accept this PR before 1.0, would accepting it later be a breaking change in any way?

Not sure but i would say no. No breaking change. It adds osgi metadata to the jar file which is irrelevant for other non-osgi users. But having it would be good for initial adoption osgi people would be grateful 😃

**Comment by bjmi** [comment]
It is very welcome if Eclipse developers can also benefit from the JSpecify annotations right from the start.
They can be used e.g. in [JDT Null analysis](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_annotations.htm) in addition to or instead of annotations from _org.eclipse.jdt.annotation.*_
This eliminates the need to subsequently repackage the artifacts with enriched OSGI information in _MANIFEST.MF_ and  
helps to distribute and promote the library as they can be used out of the box.

Would it make sense to link PR #428?
